mkosi/initrd.py

```diff
@@ -365,7 +365,10 @@ def main() -> None:
                 shutil.copytree(
                     Path("/etc") / p,
                     Path(sandbox_tree) / "etc" / p,
-                    ignore=shutil.ignore_patterns("gnupg"),
+                    # If we're running as root, use the keyring from the host. If we're not running as root,
+                    # we might not have the necessary permissions to access the keyring so don't try to copy
+                    # the keyring in that case.
+                    ignore=shutil.ignore_patterns("S.*" if os.getuid() == 0 else "gnupg"),
```

Comment:
It's not clear from looking at it what the `S.*` patter is good for, maybe add a comment?